util/java/libtiled-java/src/main/java/org/mapeditor/io/TMXMapReader.java

```diff
@@ -931,4 +935,17 @@ public static boolean checkRoot(String filename) {
     private Entry<Integer, TileSet> findTileSetForTileGID(int gid) {
         return tilesetPerFirstGid.floorEntry(gid);
     }
+
+    /**
+     * Tile map can be assembled on UNIX system, but read on Microsoft Windows system.
+     * @param path path to imageSource, tileSet, etc.
+     * @return path with the correct {@link File#separator}
+     */
+    private String replacePathSeparator(String path) {
+        if (path == null)
+            throw new IllegalArgumentException("path cannot be null.");
+        if (path.isEmpty() || path.lastIndexOf(File.separatorChar) >= 0)
+            return path;
+        return path.replace("/", File.separator);
```

Comment:
Hello! The problem occurs during the initialization of xmlPath.
Before you take a substring from the filename, you need to replace the separator. Otherwise, the xmlPath will not be initialized, or it will not be initialized correctly.